clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -270,12 +262,12 @@ public boolean partitionsAutoAssigned() {
         return !this.subscription.isEmpty();
     }
 
-    public void consumed(TopicPartition tp, long offset) {
-        assignedState(tp).consumed(offset);
+    public void position(TopicPartition tp, long offset) {
```

Comment:
Agreed, I was trying to figure out under what conditions it would be useful to allow for multiple fetch requests while you haven't asked for the data back. I can think of two cases:

1. Traffic is very bursty but you still want low latency, so you might prefer a smaller `fetch.min.bytes`, but also to be able to buffer up data so that when it's coming in fast, you'll .
2. Processing speed is variable (or messages vary significantly in size), so 1 fetch with a fixed max bytes setting won't necessarily keep you fed with enough data.

Both are probably fairly niche and you probably need to hit an extreme case to warrant adjusting consumer settings much, but I don't think we have a way to control it at the moment.